pytorch_lightning/loggers/neptune.py

```diff
@@ -46,355 +94,556 @@ class NeptuneLogger(LightningLoggerBase):
+            self._run_name = self._run_instance["sys/name"].fetch()
+        except NeptuneOfflineModeFetchException:
+            self._run_name = "offline-name"
+
+    def _init_run_instance(self, api_key, project, name, run, neptune_run_kwargs) -> Run:
+        if run is not None:
+            run_instance = run
+        else:
+            try:
+                run_instance = neptune.init(
+                    project=project,
+                    api_token=api_key,
+                    name=name,
+                    **neptune_run_kwargs,
+                )
+            except NeptuneLegacyProjectException as e:
+                raise TypeError(
+                    f"""Project {project} has not been migrated to the new structure.
+                    You can still integrate it with the Neptune logger using legacy Python API
+                    available as part of neptune-contrib package:
+                      - https://docs-legacy.neptune.ai/integrations/pytorch_lightning.html\n
+                    """
+                ) from e
+
+        # make sure that we've log integration version for both newly created and outside `Run` instances
+        run_instance[_INTEGRATION_VERSION_KEY] = __version__
+
+        # keep api_key and project, they will be required when resuming Run for pickled logger
+        self._api_key = api_key
+        self._project_name = run_instance._project_name  # skipcq: PYL-W0212
+
+        return run_instance
+
+    def _construct_path_with_prefix(self, *keys) -> str:
+        """Return sequence of keys joined by `LOGGER_JOIN_CHAR`, started with `_prefix` if defined."""
+        if self._prefix:
+            return self.LOGGER_JOIN_CHAR.join([self._prefix, *keys])
+        return self.LOGGER_JOIN_CHAR.join(keys)
+
+    @staticmethod
+    def _verify_input_arguments(
+        api_key: Optional[str],
+        project: Optional[str],
+        name: Optional[str],
+        run: Optional["Run"],
+        neptune_run_kwargs: dict,
+    ):
+
+        # check if user used legacy kwargs expected in `NeptuneLegacyLogger`
+        used_legacy_kwargs = [
+            legacy_kwarg for legacy_kwarg in neptune_run_kwargs if legacy_kwarg in _LEGACY_NEPTUNE_INIT_KWARGS
+        ]
+        if used_legacy_kwargs:
+            raise ValueError(
+                f"Following kwargs are deprecated: {used_legacy_kwargs}.\n"
+                "If you are looking for the Neptune logger using legacy Python API,"
+                " it's still available as part of neptune-contrib package:\n"
+                "  - https://docs-legacy.neptune.ai/integrations/pytorch_lightning.html\n"
+                "The NeptuneLogger was re-written to use the neptune.new Python API\n"
+                "  - https://neptune.ai/blog/neptune-new\n"
+                "  - https://docs.neptune.ai/integrations-and-supported-tools/model-training/pytorch-lightning\n"
+                "You should use arguments accepted by either NeptuneLogger.init() or neptune.init()"
```

Comment:
Could de-duplicate the code here a bit, ideally this message would only need to be defined once